### PR TITLE
[Docs][Tutorials] Update SDKs and your first coin tutorial to use aptos_account::transfer_coins

### PR DESCRIPTION
**Pull request by movekevin**
### Description
aptos_account::transfer_coins is now preferred instead of coin::transfer as it implicitly registers an account to receive CoinType if not registered. This saves an extra step/transaction.

### Test Plan
Manual run against local testnet

### CODE REVIEW

File: developer-docs-site/docs/tutorials/first-coin.md
Comment (gregnazario) [inline developer-docs-site/docs/tutorials/first-coin.md:295]: This doesn't look right?

**Review by gregnazario — CHANGES_REQUESTED** [review]
I'm blocking this until mainnet has the framework changes

File: developer-docs-site/docs/tutorials/first-coin.md
Comment (movekevin) [reply developer-docs-site/docs/tutorials/first-coin.md]: Updated.

**Comment by movekevin** [comment]
> I'm blocking this until mainnet has the framework changes

Sounds good

**Review by clay-aptos — COMMENTED** [review]
Thanks, Kevin. The docs changes LGTM except I still see references to MoonCoin in the code although you are removing its introduction from the docs. Intentional?

**Comment by movekevin** [comment]
> Thanks, Kevin. The docs changes LGTM except I still see references to MoonCoin in the code although you are removing its introduction from the docs. Intentional?

What do you mean "you are removing its introduction from the docs"? I double checked the changes and didn't see anything related.

File: ecosystem/typescript/sdk/examples/typescript/your_coin.ts
Comment (0xmaayan) [inline ecosystem/typescript/sdk/examples/typescript/your_coin.ts:43]:
```suggestion
  /** Transfers the newly created coin to a specified receiver address */
```

File: ecosystem/typescript/sdk/src/coin_client.ts
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/coin_client.ts:67]: that looks like a type of a breaking change in the underline implementation, where users don't `createReceiverIfMissing` in their app - now have to create a user..

Comment (movekevin) [reply ecosystem/typescript/sdk/src/coin_client.ts:67]: aptos_account::transfer_coins implicitly creates the account if it doesn't exist

Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/coin_client.ts:67]: correct, but the existing functionality giving the user to choose whether they want to create the account if it doesnt exist (hence the use of `0x1::coin::transfer`) - I am just saying that this decision that is now left for the user is gone, and introduce an underlying implementation change

**Review by clay-aptos — COMMENTED** [review]
Docs changes LGTM. Awaiting Eng approval.

**Comment by clay-aptos** [comment]
> > Thanks, Kevin. The docs changes LGTM except I still see references to MoonCoin in the code although you are removing its introduction from the docs. Intentional?
> 
> What do you mean "you are removing its introduction from the docs"? I double checked the changes and didn't see anything related.

This deleted line:
"In order to transfer, withdraw, or deposit coins, you must have a `CoinStore` registered for the specific coin. In this tutorial, this is `CoinStore<MoonCoin>`."

But I see we have plenty of references to MoonCoin before that so please disregard my noise.

**Review by clay-aptos — APPROVED** [review]

File: ecosystem/typescript/sdk/src/coin_client.ts
Comment (movekevin) [reply ecosystem/typescript/sdk/src/coin_client.ts]: I added createReceiverIfMissing back.

File: ecosystem/typescript/sdk/src/plugins/coin_client.ts
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/plugins/coin_client.ts:70]: where is `coinTypeToTransfer` defined?

Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/plugins/coin_client.ts:70]: nvm, needed to expand the file

**Review by 0xmaayan — APPROVED** [review]

File: ecosystem/typescript/sdk/src/plugins/coin_client.ts
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/plugins/coin_client.ts:64]: Why not just always use `transfer_coins`?  It's a minimal gas difference right?

**Review by gregnazario — APPROVED** [review]

File: ecosystem/typescript/sdk/src/plugins/coin_client.ts
Comment (movekevin) [reply ecosystem/typescript/sdk/src/plugins/coin_client.ts:64]: I really wanted to kill createReceiverIfMissing but unfortunately that'd be a breaking change (passing createReceiverIfMissing would suddenly stop doing anything). We can discuss that further separately from this change

**Pull request merged**
